File: test-suite/src/test/java/io/micronaut/mqtt/docs/parameters/ProductClient.java
package io.micronaut.mqtt.docs.parameters;

// tag::imports[]
import io.micronaut.mqtt.annotation.Topic;
import io.micronaut.mqtt.annotation.v5.MqttPublisher;
// end::imports[]

import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@MqttPublisher
public interface ProductClient {

    @Topic("product") // <1>
    void send(byte[] data);

    void send(@Topic String binding, byte[] data); // <2>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/mqtt/docs/parameters/ProductClient.kt
package io.micronaut.mqtt.docs.parameters

// tag::imports[]
import io.micronaut.mqtt.v5.annotation.MqttPublisher
import io.micronaut.context.annotation.Requires
import io.micronaut.mqtt.annotation.Topic
// end::imports[]

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@MqttPublisher
interface ProductClient {

    @Topic("product") // <1>
    fun send(data: ByteArray)

    fun send(@Topic binding: String, data: ByteArray)  // <2>
}
// end::clazz[]
